sys/net/gnrc/netif/gnrc_netif_ieee802154.c

```diff
@@ -95,7 +94,9 @@ static gnrc_pktsnip_t *_recv(gnrc_netif_t *netif)
             gnrc_pktbuf_release(pkt);
             return NULL;
         }
-        if (!(state->flags & NETDEV_IEEE802154_RAW)) {
+        netopt_enable_t raw = NETOPT_DISABLE;
+        dev->driver->get(dev, NETOPT_RAWMODE, &raw, sizeof(raw));
+        if (raw == NETOPT_DISABLE) {
```

Comment:
Mh... maybe have a flag for RAW mode in `gnrc_netif_t::flags` instead (of cause independent of IEEE 802.15.4) that is set on interface initialization and in `gnrc_netif_set_from_netdev()`. This way we may circumvent the (minimal) timing issue.

Reply:
Added the flag. I decided not to check if the flag is set on initialization of the netdev device as it really must not be enable by default in a netdev instance.